### PR TITLE
Support rails 6.1, ruby 2.7, ruby 3.0

### PR DESCRIPTION
**Pull request by rewritten**
Improvement on version support. This is also available at pyromaniac/active_data#77.

- [x] Several methods that assumed `errors` is a mutable array now fork behavior depending on the ActiveModel version.
- [x] Explicitly use kwargs splat in `method_missing` in Scopes
- [x] Wrap scope again when ruby 3.0 unwraps it (by calling array methods for instance)

References:

<https://rubyreferences.github.io/rubychanges/3.0.html#array-always-returning-array>

> On custom classes inherited from Array, some methods previously were returning an instance of this class, and others returned Array. Now they all do the latter.

<https://rubyreferences.github.io/rubychanges/2.7.html#keyword-argument-related-changes>

> Empty hash splat doesn’t pass empty hash as a positional argument.

<https://github.com/rails/rails/blob/6-1-stable/activemodel/CHANGELOG.md>

> The ActiveModel’s errors collection is now an array of these Error objects, instead of messages/details hash.

### CODE REVIEW

File: lib/active_data.rb [hunk 1]
```diff
@@ -43,6 +43,12 @@ def self.config
     ActiveData::Config.instance
   end
 
+  def self.legacy_active_model?
```
Comment (jodosha) [inline lib/active_data.rb]: @rewritten Maybe pass the definition of legacy as argument to this method.

When AM 7.0 will be released how would you scale this method if the definition of legacy is hardcoded?

File: lib/active_data/model/representation.rb [hunk 2]
```diff
@@ -67,12 +67,21 @@ def run_validations! #:nodoc:
       #
       def emerge_represented_attributes_errors!
         self.class.represented_attributes.each do |attribute|
-          key = :"#{attribute.reference}.#{attribute.column}"
+          move_errors(:"#{attribute.reference}.#{attribute.column}", attribute.column)
+        end
+      end
+
+      def move_errors(from, to)
+        if ActiveData.legacy_active_model?
```
Comment (jodosha) [inline lib/active_data/model/representation.rb]: See my previous comment, we can do:

```ruby
ActiveData.legacy_active_model?("6.1.0")
```

File: lib/active_data/model/scopes.rb [hunk 3]
```diff
@@ -37,11 +37,27 @@ def respond_to_missing?(method, _)
           super || self.class._scope_model.respond_to?(method)
         end
 
-        def method_missing(method, *args, &block)
+        def method_missing(method, *args, **kwargs, &block)
```
Comment (jodosha) [inline lib/active_data/model/scopes.rb]: @rewritten I would define multiple versions of `#method_missing`, by checking `RUBY_VERSION` constant.

The `if` statements are adding performance overhead each time this method is invoked.

Example:

```ruby
if RUBY_VERSION ~= /\A2.6/
  def method_missing # 2.6 compatible implementation
  end
elsif RUBY_VERSION ~= /\A2.7/
    def method_missing # 2.7 compatible implementation
    end
elsif RUBY_VERSION ~= /\A3/
    def method_missing # 3 compatible implementation
    end
end
```

**Review by jodosha — COMMENTED** [review]
@rewritten Thanks for taking care of this. I left a few comments, please let me know what you think.

File: lib/active_data.rb (hunk 1 above)
Comment (rewritten) [reply lib/active_data.rb]: Then it's the same as moving the gem version comparison to each of the 2 calling places (which I actually prefer)

Comment (jodosha) [reply lib/active_data.rb]: @rewritten I didn't get what you mean here. Do you have an example?

**Review by jodosha — APPROVED** [review]
@rewritten Thanks for taking care of this.

Please rebase `master` so we can run GHA on this branch as well.

**Pull request merged**